lib/components/app/session-timeout.tsx

```diff
@@ -0,0 +1,124 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+/* eslint-disable react/prop-types */
+import { Button, Modal } from 'react-bootstrap'
+import { connect } from 'react-redux'
+import { FormattedMessage } from 'react-intl'
+import React, { Component } from 'react'
+
+import * as uiActions from '../../actions/ui'
+
+/**
+ * This component makes the current session timeout
+ * by displaying a timeout warning one minute before the timeout,
+ * and by reloading the initial URL if there is no user-initiated
+ * actions within the timeout window.
+ */
+class SessionTimeout extends Component {
+  state = {
+    showTimeoutWarning: false,
+    timeoutObject: null,
+    timeoutStartMillis: 0
+  }
+
+  componentDidMount() {
+    // Wait one second or so after loading before probing for changes
+    // so that initialization actions can complete.
+    setTimeout(this.handleAfterInitialActions, 1500)
+  }
+
+  componentWillUnmount() {
+    // @ts-ignore SessionTimeout is not typed yet
```

Comment:
Instead of using `ts-ignore` with an `eslint-disable`, you can use `ts-expect-error`.

Reply:
Added types instead in https://github.com/opentripplanner/otp-react-redux/pull/574/commits/292181c4439ad16d50a1e1860fd7504c3b1f2721.